src/compute/src/render/sinks.rs

```diff
@@ -36,7 +37,7 @@ where
         tokens: &mut std::collections::BTreeMap<GlobalId, Rc<dyn std::any::Any>>,
         import_ids: BTreeSet<GlobalId>,
         sink_id: GlobalId,
-        sink: &SinkDesc,
+        sink: &SinkDesc<CollectionMetadata>,
```

Comment:
`SinkDesc` get a "storage metadata type", akin to `SourceInstanceDesc`. The coordinator deals exclusively with `SinkDesc<()>`. The controller converts these into `SinkDesc<CollectionMetadata>` before sending them to the compute instances, by asking the storage controller for the collection information.